src/utils/git.js

```diff
@@ -0,0 +1,42 @@
+import _ from 'lodash'
+import cp from 'child_process'
+import Promise from 'bluebird'
+
+export const execFile = Promise.promisify(cp.execFile)
+
+export function parseDiffRanges(diff) {
+  const matches = diff.match(/^@@ -\d+,\d+ \+(\d+),(\d+) @@/gm)
+  if (!_.isEmpty(matches)) {
+    return matches.map(match => {
+      const [start, end] = /^@@ -\d+,\d+ \+(\d+),(\d+) @@/.exec(match).slice(1, 3)
+      return [parseInt(start, 10), parseInt(start, 10) + parseInt(end, 10)]
+    })
+  }
+  return []
+}
+
+const filenameRegex = /a\/([^\n]+) b\/[^\n]+/
```

Comment:
`.` matches every character _except newline_, so `[^\n]` can be simplified to `.` which is exactly the same.